pkg/lib/userexport/model.go

```diff
@@ -226,3 +231,77 @@ func NewResponseFromTask(task *redisqueue.Task) (*Response, error) {
 
 	return response, nil
 }
+
+func jsonPointerToCSVFieldName(pointer string) (fieldName string) {
+	splitFn := func(c rune) bool {
+		return c == '/'
+	}
+	fields := strings.FieldsFunc(pointer, splitFn)
+	return strings.Join(fields, ".")
+}
+
+func ExtractCSVHeaderField(fieldPointer []*FieldPointer) (headerFields *CSVHeaderFieldNames, err error) {
+	isDuplicated := false
+	fields := make([]string, 0)
+	fieldsMap := map[string]bool{}
+	for _, pointer := range fieldPointer {
+		var fieldName string
+		if pointer.FieldName == "" {
+			fieldName = jsonPointerToCSVFieldName((pointer.Pointer))
+		} else {
+			fieldName = pointer.FieldName
+		}
+
+		if fieldsMap[fieldName] {
+			isDuplicated = true
+		}
+
+		fieldsMap[fieldName] = true
+		fields = append(fields, fieldName)
+	}
+
+	headerFields = &CSVHeaderFieldNames{
+		FieldNames: fields,
+	}
+
+	if isDuplicated {
+		headerFieldsString, _ := json.Marshal(headerFields)
+		return nil, ErrUserExportDuplicateField.New(fmt.Sprintf("%s", headerFieldsString))
```

Comment:
https://github.com/authgear/authgear-server/pull/4717/files#diff-5d0c564f0dbed14ca1655b1752d72835c8115dbe937c0a20f830584f1ff8edefR190

The spec says you need to populate the error info `field_names`
The error message should be static, use this `field names are not unique`

Reply:
updated to this message, the duplicated field moved to error info